src/utils.ts

```diff
@@ -0,0 +1,10 @@
+
+export function getClosest(elm: Element, selector: string): HTMLElement | null {
+  while (elm && elm.parentNode !== document) {
+    if (elm.matches(selector)) {
+      return elm as HTMLElement;
+    }
+    elm = elm.parentNode as HTMLElement;
+  }
+  return null;
+}
```

Comment:
This seems like a function that is generally useful for both places so I added it to phoenix.

Reply:
Yeah, i find this function very useful when working with vanilla javascript to access the DOM. If the list of functions that are not used directly by this module grows long, we might want to move them to a separate module.